src/calculations.js

```diff
@@ -354,9 +355,15 @@ export const Calculations = {
           )
         )
         this.sound?.playExplosion()
-        bodies[j].radius = 0n
+
+        body.starLvl += 1
+
+        if (body.starLvl >= body.maxStarLvl) {
+          bodies[j].radius = 0n
```

Comment:
radius should actually go to 0 when missile hits regardless of maxStarLvl since the body should still only be hit once per round. If the round is cleared then the starLvl goes up in the contract. If the starLvl == maxStarLvl then the body is "removed" from the problem in the contract and is added to an array of star trophies.